src/Data/String/CodePoints.purs

```diff
@@ -0,0 +1,273 @@
+-- | These functions allow PureScript strings to be treated as if they were
+-- | sequences of Unicode code points instead of their true underlying
+-- | implementation (sequences of UTF-16 code units). For nearly all uses of
+-- | strings, these functions should be preferred over the ones in Data.String.
+module Data.String.CodePoints
+  ( module StringReExports
+  , CodePoint()
+  , codePointAt
+  , codePointFromInt
+  , codePointToInt
+  , count
+  , drop
+  , dropWhile
+  , fromCodePointArray
+  , indexOf
+  , indexOf'
+  , lastIndexOf
+  , lastIndexOf'
+  , length
+  , singleton
+  , splitAt
+  , take
+  , takeWhile
+  , toCodePointArray
+  , uncons
+  ) where
+
+import Prelude
+
+import Data.Array as Array
+import Data.Char as Char
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.String as String
+import Data.String.Unsafe as Unsafe
+-- WARN: In order for this module to be a drop-in replacement for Data.String,
+-- this list must be updated to re-export any exports added to Data.String.
```

Comment:
What would you suggest saying here? "any exports which don't operate on indices or Chars"? Either way, whenever Data.String is changed, there is an action item, which is what I'm trying to convey.